integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonFileMetastore.scala

```diff
@@ -537,92 +482,38 @@ class CarbonFileMetastore extends CarbonMetaStore {
-   *
-   * @param timeStamp
    */
-  private def updateSchemasUpdatedTime(timeStamp: Long) {
-    tableModifiedTimeStore.put(CarbonCommonConstants.DATABASE_DEFAULT_NAME, timeStamp)
-  }
-
-  def updateAndTouchSchemasUpdatedTime() {
-    updateSchemasUpdatedTime(touchSchemaFileSystemTime())
+  private def updateSchemasUpdatedTime(tableUniqueId: String, timeStamp: Long) {
+    CarbonFileMetastore.updateTableSchemaModifiedTime(tableUniqueId, timeStamp)
   }
 
 
-  /**
-   * This method will check and create an empty schema timestamp file
-   *
-   * @return
-   */
-  private def touchSchemaFileSystemTime(): Long = {
-    val (timestampFile, timestampFileType) = getTimestampFileAndType()
-    if (!FileFactory.isFileExist(timestampFile, timestampFileType)) {
-      LOGGER.info(s"Creating timestamp file for $timestampFile")
-      FileFactory
-        .createNewFile(timestampFile,
-          timestampFileType,
-          true,
-          new FsPermission(FsAction.ALL, FsAction.ALL, FsAction.ALL))
-    }
-    FileFactory.getCarbonFile(timestampFile, timestampFileType)
-      .setLastModifiedTime(System.currentTimeMillis())
-    // since there is no guarantee that exact same set modified time returns when called
-    // lastmodified time, so better get the time from file.
-    FileFactory.getCarbonFile(timestampFile, timestampFileType)
-      .getLastModifiedTime
-  }
-
-  def checkSchemasModifiedTimeAndReloadTable(tableIdentifier: TableIdentifier): Boolean = {
-    val (timestampFile, timestampFileType) = getTimestampFileAndType()
-    var isRefreshed = false
-    if (FileFactory.isFileExist(timestampFile, timestampFileType)) {
-      val lastModifiedTime =
-        FileFactory.getCarbonFile(timestampFile, timestampFileType).getLastModifiedTime
-      if (!(lastModifiedTime ==
-            tableModifiedTimeStore.get(CarbonCommonConstants.DATABASE_DEFAULT_NAME))) {
-        metadata.writeLock.lock()
-        metadata.carbonTables = metadata.carbonTables.filterNot(
-          table => table.getTableName.equalsIgnoreCase(tableIdentifier.table) &&
-                   table.getDatabaseName
-                     .equalsIgnoreCase(tableIdentifier.database
-                       .getOrElse(SparkSession.getActiveSession.get.sessionState.catalog
-                         .getCurrentDatabase)))
-        metadata.writeLock.unlock()
-        updateSchemasUpdatedTime(lastModifiedTime)
-        isRefreshed = true
-      }
+  override def isSchemaRefreshed(tableUniqueId: String, tablePath: String): Boolean = {
+    val schemaFilePath = CarbonTablePath.getSchemaFilePath(tablePath)
+    val schemaCarbonFile = FileFactory.getCarbonFile(schemaFilePath)
+    val oldTime = Option(CarbonFileMetastore.getTableModifiedTime(tableUniqueId))
+    val newTime = schemaCarbonFile.getLastModifiedTime
+    oldTime match {
+      case Some(cacheTime) =>
+        cacheTime != newTime
+      case None => true
```

Comment:
Where you are updating the schema file time to the map if schema file is updated by another driver?

Reply:
everytime a schema file is written we are updating the map..Refer: **org.apache.spark.sql.hive.CarbonFileMetastore#createSchemaThriftFile**

Reply:
How another driver updates the latest time into the map.?